tests/extensions/image/test_exporter.py

```diff
@@ -0,0 +1,239 @@
+        format = '{}.{}'.format(settings.EXPORT_MAXIMUM_SIZE, settings.EXPORT_TYPE)
+        exporter = ImageExporter(source_file_path=source_file_path, ext='.png',
+                            output_file_path=output_file_path, format=format)
+
+        assert not os.path.exists(output_file_path)
+        assert os.path.exists(directory)
+
+        exporter.export()
+
+        assert os.path.exists(output_file_path)
+
+        output_image = Image.open(output_file_path)
+        source_image = Image.open(source_file_path)
+        source_pixels = list(source_image.getdata())
+        output_pixels = list(output_image.getdata())
+
+        assert source_image.size == output_image.size
+        assert output_image.mode == 'RGB'
+        assert output_image.palette == source_image.palette
+        assert output_image.format.lower() == settings.EXPORT_TYPE
+
+        for i in range(100):
+            # This conversion changes some pixels wildy... So high tolerance
+            assert isclose(source_pixels[i][0], output_pixels[i][0], abs_tol=20)
+            assert isclose(source_pixels[i][1], output_pixels[i][1], abs_tol=20)
+            assert isclose(source_pixels[i][2], output_pixels[i][2], abs_tol=20)
+
+    def test_png_with_transparency(self, directory):
+        source_file_path = os.path.join(BASE, 'files', 'test_transparency.png')
+        output_file_path = os.path.join(directory, 'test.{}'.format(settings.EXPORT_TYPE))
+        format = '{}.{}'.format(settings.EXPORT_MAXIMUM_SIZE, settings.EXPORT_TYPE)
+        exporter = ImageExporter(source_file_path=source_file_path, ext='.png',
+                            output_file_path=output_file_path, format=format)
+
+        assert not os.path.exists(output_file_path)
+        assert os.path.exists(directory)
+
+        exporter.export()
+
+        assert os.path.exists(output_file_path)
+
+        output_image = Image.open(output_file_path)
+        source_image = Image.open(source_file_path)
+        source_pixels = list(source_image.getdata())
+        output_pixels = list(output_image.getdata())
+
+        assert source_image.size == output_image.size
+        assert output_image.mode == 'RGB'
+        assert output_image.palette == source_image.palette
+        assert output_image.format.lower() == settings.EXPORT_TYPE
+        for i in range(100):
+            # Check if conversion is close OR if the value on the source is 0. A value on the source
+            # Being zero means it was most likely transparent, and got converted to white.
+            assert isclose(source_pixels[i][0], output_pixels[i][0], abs_tol=5) or (
+                isclose(0, source_pixels[i][0], abs_tol=1))
+            assert isclose(source_pixels[i][1], output_pixels[i][1], abs_tol=5) or (
+                isclose(0, source_pixels[i][1], abs_tol=1))
+            assert isclose(source_pixels[i][2], output_pixels[i][2], abs_tol=5) or (
+                isclose(0, source_pixels[i][2], abs_tol=1))
```

Comment:
Reorder asserts so it is either transparent or close to source.

Reply:
complete